dlc-node/src/main/scala/org/bitcoins/dlc/node/DLCServer.scala

```diff
@@ -0,0 +1,98 @@
+package org.bitcoins.dlc.node
+
+import akka.actor._
+import akka.event.LoggingReceive
+import akka.io.{IO, Tcp}
+import org.bitcoins.core.api.dlc.wallet.DLCWalletApi
+import org.bitcoins.dlc.node.config._
+import org.bitcoins.tor.TorController
+
+import java.io.IOException
+import java.net.InetSocketAddress
+import scala.concurrent.{Future, Promise}
+
+class DLCServer(
+    dlcWalletApi: DLCWalletApi,
+    bindAddress: InetSocketAddress,
+    boundAddress: Option[Promise[InetSocketAddress]],
+    dataHandlerFactory: DLCDataHandler.Factory = DLCDataHandler.defaultFactory)
+    extends Actor
+    with ActorLogging {
+
+  import context.system
+
+  IO(Tcp) ! Tcp.Bind(self, bindAddress)
+
+  override def receive: Receive = LoggingReceive {
+    case Tcp.Bound(localAddress) =>
+      log.info(s"Bound at $localAddress")
+      boundAddress.foreach(_.success(localAddress))
+
+    case c @ Tcp.CommandFailed(_: Tcp.Bind) =>
+      val ex = c.cause.getOrElse(new IOException("Unknown Error"))
+      log.error(s"Cannot bind $boundAddress", ex)
+      throw ex
+
+    case Tcp.Connected(remoteAddress, _) =>
+      val connection = sender()
+      log.info(s"Received a connection from $remoteAddress")
+      val _ = context.actorOf(
+        Props(
+          new DLCConnectionHandler(dlcWalletApi,
+                                   connection,
+                                   None,
+                                   dataHandlerFactory)))
+  }
+
+  override def aroundReceive(receive: Receive, msg: Any): Unit = try {
```

Comment:
This is an internal API to akka. Is there a need for overriding this? 

https://github.com/akka/akka/blob/f3fb2a577d804fb69e6b1ff2fd8f3774627ad81f/akka-actor/src/main/scala/akka/actor/Actor.scala#L535

Reply:
this was from @rorp's code, maybe he can answer

Reply:
If we want `try/catch` in all `Receive` methods then there's no need for overriding it.